src/main/java/com/mate/academy/demo/repository/BookDaoImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.mate.academy.demo.repository;
+
+import com.mate.academy.demo.exception.DataProcessingException;
+import com.mate.academy.demo.model.Book;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookDaoImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookDaoImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public Book save(Book book) {
+        Transaction transaction = null;
+        try (Session session = sessionFactory.openSession()) {
```

Comment:
don't use try with resources and transaction, session will close before catch block